### PR TITLE
feature/dropUp

### PR DESCRIPTION
**Pull request by PanzerMessiaH**
Added ability dor dropdown to 'drop up' when the dropdown is out of view.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,14 +20,19 @@ import keyboardNavigation from './tree-manager/keyboardNavigation'
...
 
+import { findDOMNode } from 'react-dom'
+
 const cx = cn.bind(styles)
 
 const initialState = {
   searchInput: '',
   searchModeOn: false,
   currentFocus: undefined,
+  dropUp: false,
```
Comment (robyn-thomas) [inline src/index.js]: Please remove comma here

File: src/index.js [hunk 2]
```diff
@@ -61,6 +66,7 @@ class DropdownTreeSelect extends Component {
     enforceSingleSelection: PropTypes.bool,
     value: PropTypes.string,
     onInputChange: PropTypes.func,
+    autoDetermineDropUp: PropTypes.bool,
```
Comment (robyn-thomas) [inline src/index.js]: And here (the comma)

File: src/index.js [hunk 3]
```diff
@@ -342,6 +358,54 @@ class DropdownTreeSelect extends Component {
...
+
+    let dropUp = false
+    if (inputBoxRect.bottom + nodeBoundingRect.height > windowHeight) {
+      dropUp = inputBoxRect.top - nodeBoundingRect.height > WINDOW_OFFSET
+    }
+
+    this.setState({
+      dropUp,
+    })
+  }, 250)
```
Comment (robyn-thomas) [inline src/index.js]: Can we make this 250 a const variable?

File: src/index.js (hunk 1 above)
Comment (PanzerMessiaH) [reply src/index.js]: Comma needs to be there, this libraries rules are setup to give errors when its not there.

File: src/index.js (hunk 2 above)
Comment (PanzerMessiaH) [reply src/index.js]: Same here.

File: src/index.js (hunk 3 above)
Comment (PanzerMessiaH) [reply src/index.js]: I can yes.

**Pull request merged**